packages/audioplayers_linux/linux/audio_player.cc

```diff
@@ -0,0 +1,317 @@
+#include "audio_player.h"
+
+#include <flutter_linux/flutter_linux.h>
+
+#include "Logger.h"
+
+AudioPlayer::AudioPlayer(std::string playerId, FlMethodChannel *channel)
+    : _playerId(playerId), _channel(channel) {
+    gst_init(NULL, NULL);
+    playbin = gst_element_factory_make("playbin", "playbin");
+    if (!playbin) {
+        Logger::Error(std::string("Not all elements could be created."));
+        return;
+    }
+
+    // Setup source options
+    g_signal_connect(playbin, "source-setup",
+                     G_CALLBACK(AudioPlayer::SourceSetup), &source);
+
+    bus = gst_element_get_bus(playbin);
+
+    // Watch bus messages for one time events
+    gst_bus_add_watch(bus, (GstBusFunc)AudioPlayer::OnBusMessage, this);
+
+    // Refresh continuously to emit reoccuring events
+    g_timeout_add(1000, (GSourceFunc)AudioPlayer::OnRefresh, this);
+}
+
+AudioPlayer::~AudioPlayer() {}
+
+void AudioPlayer::SourceSetup(GstElement *playbin, GstElement *source,
+                              GstElement **p_src) {
+    // Allow sources from unencrypted / misconfigured connections
+    if (g_object_class_find_property(G_OBJECT_GET_CLASS(source),
+                                     "ssl-strict") != 0) {
+        g_object_set(G_OBJECT(source), "ssl-strict", FALSE, NULL);
+    }
+};
+
+void AudioPlayer::SetSourceUrl(std::string url) {
+    if (_url != url) {
+        _url = url;
+        if (_url.empty()) {
+            gst_element_set_state(playbin, GST_STATE_NULL);
+        } else {
+            g_object_set(playbin, "uri", _url.c_str(), NULL);
+            if (playbin->current_state != GST_STATE_READY)
+                gst_element_set_state(playbin, GST_STATE_READY);
```

Comment:
```suggestion
                gst_element_set_state(playbin, GST_STATE_READY);
```

Some brackets would be nice